### ISSUE TITLE
[Feature]: datasource linode_instance_types

### ISSUE THREAD

**Issue by elreydetoda** [linode/terraform-provider-linode#501]
### Description

Something that I normally do, is explore data sources when I am looking at a terraform provider. When doing that, I normally explore the necessary datasources that I need for constructing a resource. I was going to try and deploy a `linode_instances` resource, and when looking at the docs it showed that I would need an `instance_type`. Unfortunately it doesn't appear to exist `instance_types` data source ( from my understanding this would be analogous to DigitalOcean's (DO) [droplets](https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/data-sources/droplets) terraform data source.

So, ideally there would be an `instance_types` module, and then you can specify optional tags ( similar to the `linode_images` ). That way you could return all instance_types and can filter down as necessary.

### New or Affected Terraform Resources

instance_types

### Potential Terraform Configuration

```terraform
data "linode_instance_types" "all" {}
```

or optionally apply a filter:

```terraform
data "linode_instance_types" "all" {
  filter {
    key    = "vcpus"
    values = ["1"]
  }
}

**Comment by LBGarber** [comment]
Hi there!
This is a great idea! I've put up a PR to add this functionality and it should be available in release v1.22.0.

**Comment by elreydetoda** [comment]
Awesome, thank you @LBGarber :grin:
